src/main/java/org/opensearch/ad/ml/CheckpointDao.java

```diff
@@ -425,6 +492,52 @@ private void logFailure(BulkByScrollResponse response, String detectorID) {
         }
     }
 
+    private RandomCutForest restoreRCFModel(String rcfCheckpoint) {
+        if (Strings.isEmpty(rcfCheckpoint)) {
+            return null;
+        }
+
+        String checkpoint = rcfCheckpoint.trim();
+        RandomCutForest forest = null;
+        try {
+            if (checkpoint.charAt(0) == '{' && checkpoint.charAt(checkpoint.length() - 1) == '}' && isJson(checkpoint)) {
```

Comment:
Make this a "isVersionEqualsOne?" function? Again not urgent - but we may have more use for that function later.

Reply:
yes, changed